### PR TITLE
Fix warning messages on strip_column_names

### PR DESCRIPTION
**Pull request by jburos**
Cleaned up tests in order to fix printed warning message from `test/test_utils.py`. This PR should resolve issue #89.

Turned out this was not due to problems in global warning/logging strategy. Instead, there was one intentional warning in a test case which I had forgotten to capture. 

In the course of cleaning this up, I decided to comment-out the earlier & now redundant use of `logging` & to update the test cases to confirm that warnings are created.

### CODE REVIEW

File: cohorts/utils.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def strip_column_names(cols, keep_paren_contents=True):
     > df2 = df.rename(columns = strip_column_names(df.columns,
         keep_paren_contents=False))
     """
-    logger = logging.getLogger()
```
Comment (tavinathanson) [inline cohorts/utils.py:124]: Remove comment?

File: cohorts/load.py [hunk 2]
```diff
@@ -1297,6 +1297,10 @@ def compare_provenance(
     -----------
     Number of discrepancies (0: None)
     """
+    ## if either this or other items is null, return 0
```
Comment (tavinathanson) [inline cohorts/load.py:1300]: I guess you'll merge in the other PR and update this one, and this won't be here?

File: cohorts/utils.py [hunk 3]
```diff
@@ -132,8 +132,8 @@ def strip_column_names(cols, keep_paren_contents=True):
         warn_str = 'Warning: strip_column_names (if run) would introduce duplicate names.'
         warn_str += ' Reverting column names to the original.'
 
-        logger.warning(warn_str)
-        warnings.warn(warn_str)
+        #logger.warning(warn_str)
```
Comment (tavinathanson) [inline cohorts/utils.py]: Remove comment?

File: test/test_utils.py [hunk 4]
```diff
@@ -143,7 +146,9 @@ def test_as_dataframe_bad_rename():
     df_hello, cohort = prep_test_cohort()
     # test behavior with rename_cols=True. should raise a warning
     with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
         df = cohort.as_dataframe(rename_cols=True, join_with='hello')
+        ok_(len(w)>0)
```
Comment (tavinathanson) [inline test/test_utils.py]: Style nit: `len(w) > 0`

File: test/test_utils.py [hunk 5]
```diff
@@ -152,7 +157,9 @@ def test_as_dataframe_drop_parens():
     df_hello, cohort = prep_test_cohort()
     # test behavior with keep_paren_contents=False
     with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
         df = cohort.as_dataframe(rename_cols=True, keep_paren_contents=False, join_with='hello')
+        ok_(len(w)>0)
```
Comment (tavinathanson) [inline test/test_utils.py]: Style nit: `len(w) > 0`

**Comment by jburos** [comment]
sorry - just realized i branched these changes off of my other `fix-summary` branch so this PR includes out of date versions of those changes. 

Closing for now (I'll fix & resubmit) so as to avoid confusion.

File: test/test_utils.py [hunk 6]
```diff
@@ -69,12 +69,15 @@ def test_strip_column_names():
...
     ok_((df2.columns != df.columns).any())
 
     # should not rename columns -- should error
-    df3 = df.rename(columns=strip_column_names(
-        df.columns, keep_paren_contents=False))
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        df3 = df.rename(columns=strip_column_names(
+                        df.columns, keep_paren_contents=False))
+        ok_(len(w)>0)
```
Comment (tavinathanson) [inline test/test_utils.py]: Same style nit

File: test/test_provenance.py [hunk 7]
```diff
@@ -118,6 +120,13 @@ def test_summarize_provenance():
             ok_(not cohort.summarize_provenance_per_cache()[cache_name])
             ok_(len(w)>0)
 
+        ## now test whether deleting contents of cache_dir causes an error 
+        dirlist = [ f for f in os.listdir(cache_dir) ]
+        for dir in dirlist:
+            rmtree(path.join(cache_dir, dir))
+        summary = cohort.summarize_provenance()
+
```
Comment (tavinathanson) [inline test/test_provenance.py]: Style nit: extra newline

Comment (tavinathanson) [inline test/test_provenance.py]: Oops, I had already mentioned that in the last PR!

**Comment by jburos** [comment]
@tavinathanson thanks for feedback! i'll fix the various nits & then resubmit once the earlier PR has been merged

**Comment by tavinathanson** [comment]
@jburos cool, I'm done reviewing so feel free to merge in the new PR

**Pull request merged**